core/src/test/java/org/infinispan/distribution/RemoteGetFailureTest.java

```diff
@@ -0,0 +1,240 @@
+      // cache stop takes quite long when the view splits
+      builder.clustering().stateTransfer().timeout(10, TimeUnit.SECONDS);
+      createCluster(builder, 3);
+      waitForClusterToForm();
+      key = getKeyForCache(cache(1), cache(2));
+   }
+
+   @AfterMethod(alwaysRun = true)
+   @Override
+   protected void clearContent() throws Throwable {
+      // do not clear the caches
+      TestingUtil.killCacheManagers(false, cacheManagers.toArray(new EmbeddedCacheManager[cacheManagers.size()]));
+      cacheManagers.clear();
+   }
+
+   public void testExceptionFromBothOwners(Method m) {
+      initAndCheck(m);
+
+      cache(1).getAdvancedCache().getAsyncInterceptorChain().addInterceptor(new FailingInterceptor(), 0);
+      cache(2).getAdvancedCache().getAsyncInterceptorChain().addInterceptor(new FailingInterceptor(), 0);
+
+      try {
+         cache(0).get(key);
+         fail("Should have thrown");
+      } catch (RemoteException e) {
+         assertEquals(e.getCause().getMessage(), "Injected");
+      }
+   }
+
+   public void testExceptionFromOneOwner(Method m) {
+      initAndCheck(m);
+
+      CountDownLatch release = new CountDownLatch(1);
+      cache(1).getAdvancedCache().getAsyncInterceptorChain().addInterceptor(new FailingInterceptor(), 0);
+      cache(2).getAdvancedCache().getAsyncInterceptorChain().addInterceptor(new DelayingInterceptor(null, release), 0);
+
+      // It's not enough to test if the exception is TimeoutException as we want the remote get fail immediately
+      // upon exception.
+
+      // We cannot mock TimeService in ScheduledExecutor, so we have to measure if the response was fast
+      // remoteTimeout is gracious enough (15s) to not cause false positives
+      long requestStart = System.nanoTime();
+      try {
+         cache(0).get(key);
+         fail("Should have thrown");
+      } catch (RemoteException e) {
+         long exceptionThrown = System.nanoTime();
+         long remoteTimeout = cache(0).getCacheConfiguration().clustering().remoteTimeout();
+         long delay = TimeUnit.NANOSECONDS.toMillis(exceptionThrown - requestStart);
+         assertTrue(delay < remoteTimeout);
+         assertEquals(e.getCause().getMessage(), "Injected");
+      } finally {
+         release.countDown();
+      }
+   }
+
+   public void testBothOwnersSuspected(Method m) throws ExecutionException, InterruptedException {
+      initAndCheck(m);
+
+      CountDownLatch arrival = new CountDownLatch(2);
+      CountDownLatch release = new CountDownLatch(1);
+      cache(1).getAdvancedCache().getAsyncInterceptorChain().addInterceptor(new DelayingInterceptor(arrival, release), 0);
+      cache(2).getAdvancedCache().getAsyncInterceptorChain().addInterceptor(new DelayingInterceptor(arrival, release), 0);
+
+      Future<?> future = fork(() -> {
+         try {
+            cache(0).get(key);
+            fail("Should have thrown");
+         } catch (RpcException e) {
+            // expected
```

Comment:
You're missing out on the wonderful `Exceptions` class, which allows you to write

      Future<?> future = fork(() -> cache(0).get(key));
      Exceptions.expectExecutionException(RpcException.class, future);